crates/solidity/inputs/schema/grammar/topics/05-expressions/01-base-expressions.yml

```diff
@@ -1,123 +1,276 @@
+          model: "BinaryLeftAssociative"
+          choice:
+            - terminal: "+"
+            - terminal: "-"
+
+        - name: "MulDivModExpression"
+          model: "BinaryLeftAssociative"
+          choice:
+            - terminal: "*"
+            - terminal: "/"
+            - terminal: "%"
+
+        - name: "ExponentiationExpression"
+          model: "BinaryLeftAssociative"
+          terminal: "**"
+
+        - name: "UnarySuffixExpression"
+          model: "UnarySuffix"
+          choice:
+            - terminal: "++"
+            - terminal: "--"
+
+        - name: "UnaryPrefixExpression"
+          model: "UnaryPrefix"
+          choice:
+            - terminal: "++"
+            - terminal: "--"
+            - terminal: "!"
+            - terminal: "~"
+            - terminal: "-"
+
+        - name: "FunctionCallExpression"
+          model: "UnarySuffix"
+          sequence:
+            - optional:
+                reference: "NamedArgumentList"
+            - reference: "ArgumentList"
+
+        - name: "MemberAccessExpression"
+          model: "UnarySuffix"
+          sequence:
+            - terminal: "."
+            - choice:
+                - reference: "Identifier"
+                - terminal: "address"
+
+        - name: "IndexAccessExpression"
+          model: "UnarySuffix"
+          delimitedBy:
+            open: "["
+            expression:
+              choice:
+                - sequence:
+                    - reference: "Expression"
+                    - optional:
+                        sequence:
+                          - terminal: ":"
+                          - optional:
+                              reference: "Expression"
+                - sequence:
+                    - terminal: ":"
+                    - optional:
+                        reference: "Expression"
+            close: "]"
+
+      primaryExpressions:
         - reference: "PrimaryExpression"
+
     0.6.0:
-      parserType: "Precedence"
-      choice:
-        - reference: "AssignmentExpression"
-        - reference: "ConditionalExpression"
-        - reference: "OrExpression"
-        - reference: "AndExpression"
-        - reference: "EqualityComparisonExpression"
-        - reference: "OrderComparisonExpression"
-        - reference: "BitOrExpression"
-        - reference: "BitXOrExpression"
-        - reference: "BitAndExpression"
-        - reference: "ShiftExpression"
-        - reference: "AddSubExpression"
-        - reference: "MulDivModExpression"
-        - reference: "ExponentiationExpression"
-        - reference: "UnarySuffixExpression"
-        - reference: "UnaryPrefixExpression"
-        - reference: "FunctionCallExpression"
-        - reference: "MemberAccessExpression"
-        - reference: "IndexAccessExpression"
-        - reference: "PrimaryExpression"
+      operators:
+        - name: "AssignmentExpression"
+          model: "BinaryLeftAssociative"
+          choice:
```

Comment:
To eliminate redundancy, are we able to extract these operator lists into a common top-level `Production`, so that they are not repeated between versions? It also makes it easier to find out what changed between them.

Happy to do it in a later PR if the schema changes in this PR allows it.

Reply:
* https://github.com/NomicFoundation/slang/issues/374